library.lua

```diff
@@ -174,6 +174,21 @@ function _.sort(t)
   return _.sort_by(t, _.id)
 end
 
+function _.size(t)
+  local typ = type(t)
+  if typ == "table" then
+    local size = 0
+    for k,v in pairs(t) do
+      size = size + 1
+    end
+    return size
+  elseif typ == "string" then
+    return string.len(t)
```

Comment:
Why `string.length` instead of `#`? One possibility here would be to use `#` for anything which isn't a table, though I guess that's up for debate (say if people override the number metatable).

Reply:
Then maybe something like...
```Lua
if type(t) == "table" then
  -- ...
else
  if getmetatable(t).__len then
    return #t
  else
    error("cannot get size of " .. type(t))
  end
end
```
?

Reply:
That'll fail on standard CC (non-Cobalt) due to the `getmetatable` call (also on anything with a nil metatable). Could you just rely on the fact that `#` will error (and provide somewhat helpful messages anyway).